events/logger/logger.go

```diff
@@ -118,11 +119,16 @@ func dataRequestAPIToString(
 		b.WriteString(fmt.Sprintf("Error: %s\n", data.Error))
 	}
 	if data.Response != nil {
+		tmpBody := ""
+		// do not print binary data
+		if !slices.Contains(data.Response.Header.Values("Content-Type"), "application/octet-stream") {
```

Comment:
[small] if we'll still have issues or the responses are not correctly marked as binary, maybe `utf8.validate` can be used.

Reply:
How expensive is `utf8.validation`? Extra validation/processing wastes resources - is it worth doing it here? How many cases you know with not correctly marked binary in http response?